src/Symfony/Component/Security/Csrf/TokenStorage/CookieTokenStorage.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Csrf\TokenStorage;
+
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\Security\Csrf\Exception\TokenNotFoundException;
+
+/**
+ * Accesses tokens in a set of cookies. A changeset records edits made to
+ * tokens. The changeset can be retrieved as a list of cookies to be used in a
+ * response's headers to "persist" the changes.
+ *
+ * @author Oliver Hoff <oliver@hofff.com>
+ */
+class CookieTokenStorage implements TokenStorageInterface
+{
+    /**
+     * @var array
+     */
+    private $transientTokens;
+
+    /**
+     * @var ParameterBag
+     */
+    private $cookies;
+
+    /**
+     * @var boolean
+     */
+    private $secure;
+
+    /**
+     * @param ParameterBag $cookies
+     * @param boolean $secure
+     */
+    public function __construct(ParameterBag $cookies, $secure)
+    {
+        $this->transientTokens = [];
+        $this->cookies = $cookies;
+        $this->secure = (bool) $secure;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @see \Symfony\Component\Security\Csrf\TokenStorage\TokenStorageInterface::getToken()
+     */
+    public function getToken($tokenId)
+    {
+        $token = $this->resolveToken($tokenId);
+
+        if (!strlen($token)) {
+            throw new TokenNotFoundException;
+        }
+
+        return $token;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @see \Symfony\Component\Security\Csrf\TokenStorage\TokenStorageInterface::hasToken()
+     */
+    public function hasToken($tokenId)
+    {
+        return strlen($this->resolveToken($tokenId)) > 0;
```

Comment:
`false === (bool) '0'` or do you mean casting the strlen to bool? latter would work, but than again, i thing "the length of the string is greater than zero" is so much more explicit, especially since casting from string to something else is very adventurous in PHP.